search-service/src/test/kotlin/com/egm/stellio/search/web/TemporalEntityHandlerPaginationTests.kt

```diff
@@ -0,0 +1,175 @@
+package com.egm.stellio.search.web
+
+import arrow.core.Either
+import arrow.core.right
+import com.egm.stellio.search.config.SearchProperties
+import com.egm.stellio.shared.util.BEEHIVE_TYPE
+import com.egm.stellio.shared.util.loadAndExpandSampleData
+import com.egm.stellio.shared.util.toNgsiLdFormat
+import io.mockk.coEvery
+import io.mockk.coVerify
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Test
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.TestPropertySource
+import java.time.ZonedDateTime
+
+@ActiveProfiles("test")
+@WebFluxTest(TemporalEntityHandler::class)
+@EnableConfigurationProperties(SearchProperties::class)
+@TestPropertySource(properties = ["application.pagination.temporal-limit = 5"])
+class TemporalEntityHandlerPaginationTests : TemporalEntityHandlerTestCommon() {
+
+    private val timeAt = "2019-01-01T00:00:00Z"
+    private val endTimeAt = "2021-01-01T00:00:00Z"
+    private val range = ZonedDateTime.parse(timeAt) to ZonedDateTime.parse(endTimeAt)
+    private val lastN = "100"
+    private val entityId = "urn:ngsi-ld:BeeHive:TESTC"
+
+    private suspend fun mockQueryEntitiesService(range: Range? = null) {
+        val firstTemporalEntity = loadAndExpandSampleData(
+            "/temporal/beehive_create_temporal_entity.jsonld"
+        )
+        val secondTemporalEntity = loadAndExpandSampleData("beehive.jsonld")
+        coEvery { authorizationService.computeAccessRightFilter(any()) } returns { null }
+        coEvery {
+            queryService.queryTemporalEntities(any(), any())
+        } returns Either.Right(
+            Triple(
```

Comment:
Here you are already mocking the return of the `queryTemporalEntities()` then how are you supposed to test it down below in the other tests ?

Reply:
On the test below i am not testing queryTemporalEntities but that if the queryTemporalEntities returned a range (meaning that the pagination has been trigerred). We retrurn the good status and Content-Range.
I also test that the temporalEntityQuery (with lastN and instanceLimit) is well initialize.